### PR TITLE
add type hints for mmcv/ops/...

### PR DESCRIPTION
**Pull request by zengxiang68**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Add type hints for mmcv/ops/cc_attention.py, mmcv/ops/contour_expand.py and mmcv/ops/convex_iou.py

## Modification

Only the above three files were modified. ps: I deleted "label = label.tolist()" in contour_expand.py otherwise mypy report error: "List[Any]" has no attribute "tolist".

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

**Before PR**:

- [x] I have read and followed the workflow indicated in the [CONTRIBUTING.md](https://github.com/open-mmlab/mmcv/blob/master/CONTRIBUTING.md) to create this PR.
- [x] Pre-commit or linting tools indicated in [CONTRIBUTING.md](https://github.com/open-mmlab/mmcv/blob/master/CONTRIBUTING.md) are used to fix the potential lint issues.
- [x] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] New functionalities are covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [ ] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [ ] If the modification has potential influence on downstream or other related projects, this PR should be tested with some of those projects, like MMDet or MMCls.
- [ ] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

**Comment by zhouzaida** [comment]
> [CLA assistant check](https://cla-assistant.io/open-mmlab/mmcv?pullRequest=1987) Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/open-mmlab/mmcv?pullRequest=1987) before we can accept your contribution.You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/open-mmlab/mmcv?pullRequest=1987) it.

Hi, thanks for your contribution. Please sign the CLA.

File: mmcv/ops/convex_iou.py [hunk 1]
```diff
@@ -1,10 +1,15 @@
...
+import torch
+
 from ..utils import ext_loader
 
 ext_module = ext_loader.load_ext('_ext', ['convex_iou', 'convex_giou'])
 
 
-def convex_giou(pointsets, polygons):
+def convex_giou(pointsets: torch.Tensor,
+                polygons: torch.Tensor) -> Tuple[torch.Tensor, torch.Tensor]:
```
Comment (zhouzaida) [inline mmcv/ops/convex_iou.py:12]:
```suggestion
def convex_giou(pointsets: Tensor, polygons: Tensor) -> Tuple[Tensor, Tensor]:
```

File: mmcv/ops/convex_iou.py [hunk 2]
```diff
@@ -1,10 +1,15 @@
 # Copyright (c) OpenMMLab. All rights reserved.
+from typing import Tuple
+
+import torch
```
Comment (zhouzaida) [inline mmcv/ops/convex_iou.py:4]: from torch import Tensor

File: mmcv/ops/convex_iou.py [hunk 3]
```diff
@@ -26,7 +31,8 @@ def convex_giou(pointsets, polygons):
     return convex_giou, points_grad
 
 
-def convex_iou(pointsets, polygons):
+def convex_iou(pointsets: torch.Tensor,
+               polygons: torch.Tensor) -> torch.Tensor:
```
Comment (zhouzaida) [inline mmcv/ops/convex_iou.py:35]:
```suggestion
def convex_iou(pointsets: Tensor, polygons: Tensor) -> Tensor:
```

File: mmcv/ops/convex_iou.py (hunk 2 above)
Comment (zhouzaida) [inline mmcv/ops/convex_iou.py:4]:
```suggestion
from torch import Tensor
```

File: mmcv/ops/contour_expand.py [hunk 4]
```diff
@@ -42,7 +45,6 @@ def contour_expand(kernel_mask, internal_kernel_label, min_kernel_area,
                 internal_kernel_label,
                 min_kernel_area=min_kernel_area,
                 kernel_num=kernel_num)
-            label = label.tolist()
```
Comment (zhouzaida) [inline mmcv/ops/contour_expand.py:45]: This variable `label` is Tensor here.

label = label.tolist()  # type: ignore

Comment (zhouzaida) [reply mmcv/ops/contour_expand.py:45]: It was an error from mmcv master.

You can merge the upstream master to fix the CI with the following steps.

```bash
git remote add open-mmlab git@github.com:open-mmlab/mmcv.git
git pull open-mmlab master
git push origin ytzhao/type_hints_runner_builder
```

**Review by zhouzaida — APPROVED** [review]
